lib/objects/objectsInMemFileDB.js

```diff
@@ -0,0 +1,2301 @@
+/**
+ *      Object DB in memory - Server
+ *
+ *      Copyright 2013-2018 bluefox <dogafox@gmail.com>
+ *
+ *      MIT License
+ *
+ */
+
+/* jshint -W097 */
+/* jshint strict: false */
+/* jslint node: true */
+/* jshint -W061 */
+'use strict';
+
+const extend            = require('node.extend');
+const fs                = require('fs');
+const InMemoryFileDB    = require('../inMemFileDB');
+const tools             = require('../tools.js');
+const utils             = require('./objectsUtils');
+
+/** @class */
+class ObjectsInMemoryFileDB extends InMemoryFileDB {
+
+    constructor(settings) {
+        settings.fileDB = {
+            fileName: 'objects.json',
+            backupDirName: 'backup-objects'
+        };
+        super(settings);
+
+        if (!this.change) {
+            this.change = (id /*, obj */) => {
+                this.log.silly(this.namespace + ' objects change: ' + id + ' ' + JSON.stringify(this.change));
+            };
+        }
+        this.fileOptions = {};
+        this.files = {};
+        this.writeTimer = null;
+        this.writeIds = [];
+        this.preserveSettings = [];
+        this.defaultNewAcl = this.settings.defaultNewAcl || null;
+        this.namespace = this.settings.namespace || this.settings.hostname || '';
+
+        this.objectsDir = this.dataDir + 'files/';
+
+        /*function prepareRights(options) {
+         let fOptions = {};
+         options = options || {};
+         if (!options.user) {
+         options = {
+         user: SYSTEM_ADMIN_USER,
+         params: options
+         };
+         }
+
+         // acl.owner = user this creates or owns the file
+         // acl.group = group, this assigned to file
+         // acl.permissions = '0777' - default 1 (execute, 2 write, 4 read
+         if (!options.user) {
+         fOptions.acl = {
+         owner:      SYSTEM_ADMIN_USER,
+         ownerGroup: SYSTEM_ADMIN_GROUP,
+         permissions: 0x644 // '0777'
+         };
+         } else {
+         fOptions.acl = {
+         owner: options.user
+         };
+         fOptions.acl.ownerGroup  = options.group;
+         fOptions.acl.permissions = 0x644;
+         }
+         fOptions.acl.ownerGroup  = fOptions.acl.ownerGroup || SYSTEM_ADMIN_GROUP;
+
+         return fOptions;
+         }*/
+
+        // init default new acl
+        if (this.dataset['system.config'] && this.dataset['system.config'].common && this.dataset['system.config'].common.defaultNewAcl) {
+            this.defaultNewAcl = JSON.parse(JSON.stringify(this.dataset['system.config'].common.defaultNewAcl));
+        }
+    }
+
+    // -------------- FILE FUNCTIONS -------------------------------------------
+    // memServer specific function
+    mkpathSync(rootpath, dirpath) {
```

Comment:
I think we're using `fs-extra` somewhere already. So we could replace this method with `ensureDir[Sync]`

Reply:
js.controller do not have fs-extra as dep